modules/security/src/test/java/org/apache/ignite/internal/processor/security/cache/closure/EntryProcessorRemoteSecurityContextCheckTest.java

```diff
@@ -0,0 +1,206 @@
+
+    /** Name of client initiator node. */
+    private static final String CLNT_INITIATOR = "clnt_initiator";
+
+    /** Name of server feature call node. */
+    private static final String SRV_FEATURE_CALL = "srv_feature_call";
+
+    /** Name of server feature transit node. */
+    private static final String SRV_FEATURE_TRANSITION = "srv_feature_transition";
+
+    /** Name of server endpoint node. */
+    private static final String SRV_ENDPOINT = "srv_endpoint";
+
+    /** Name of client endpoint node. */
+    private static final String CLNT_ENDPOINT = "clnt_endpoint";
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        startGrid(SRV_INITIATOR, allowAllPermissionSet());
+
+        startGrid(CLNT_INITIATOR, allowAllPermissionSet(), true);
+
+        startGrid(SRV_FEATURE_CALL, allowAllPermissionSet());
+
+        startGrid(SRV_FEATURE_TRANSITION, allowAllPermissionSet());
+
+        startGrid(SRV_ENDPOINT, allowAllPermissionSet());
+
+        startGrid(CLNT_ENDPOINT, allowAllPermissionSet(), true);
+
+        G.allGrids().get(0).cluster().active(true);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void setupVerifier(Verifier verifier) {
+        verifier
+            .add(SRV_FEATURE_CALL, 1)
+            .add(SRV_FEATURE_TRANSITION, 1)
+            .add(SRV_ENDPOINT, 1)
+            .add(CLNT_ENDPOINT, 1);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void test() {
+        runAndCheck(grid(SRV_INITIATOR));
+        runAndCheck(grid(CLNT_INITIATOR));
+    }
+
+    /**
+     * @param initiator Node that initiates an execution.
+     */
+    private void runAndCheck(IgniteEx initiator) {
+        UUID secSubjectId = secSubjectId(initiator);
+
+        for (InvokeMethodEnum ime : InvokeMethodEnum.values()) {
+            runAndCheck(
+                secSubjectId,
+                () -> compute(initiator, nodeId(SRV_FEATURE_CALL)).broadcast(
+                    new IgniteRunnable() {
+                        @Override public void run() {
+                            register();
+
+                            invoke(ime, Ignition.localIgnite(),
+                                new TestEntryProcessor(endpoints()), prmKey(grid(SRV_FEATURE_TRANSITION)));
+                        }
+                    }
+                )
+            );
+        }
+    }
+
+    /**
+     * @return Collection of endpont nodes ids.
+     */
+    private Collection<UUID> endpoints() {
+        return Arrays.asList(
+            nodeId(SRV_ENDPOINT),
+            nodeId(CLNT_ENDPOINT)
+        );
+    }
+
+    /**
+     * @param ime Invoke Method.
+     * @param node Node.
+     * @param ep Entry Processor.
+     * @param key Key.
+     */
+    private void invoke(InvokeMethodEnum ime, Ignite node,
+        EntryProcessor<Integer, Integer, Object> ep, Integer key) {
+        switch (ime) {
```

Comment:
Let's use `List<BiConsumer<EntryProcessor<Integer, Integer, Object> ep, Integer key>` where each consumer will call a specific method instead of the switch.

Reply:
We can use consumers for certain invoke methods but it would be bulky because we must use here only classes rather then lambdas.

Reply:
done